Getting-Started/Code/Source-Control/index.md

```diff
@@ -0,0 +1,139 @@
+ - **App_Data**, you can source control the generated files, including them within your Visual Studio Solution means they will be built into your application's dll
+ - **Dll**, if your working with a team, and commiting the generated dll, then this can be problematic managing merge conflicts, so remember to 'regenerate' your dll, after pulling from source control... but probably you want to use API...
+ - **API**, generate the models into a seperate extension project and source control the files from there.
+
+#### Media
+
+The Media section of Umbraco (unless configured otherwise) stores files in the /media folder, these can be updated by editors, in the Umbraco backoffice, so generally speaking you would not source control these files.
+
+#### Packages and Plugins
+
+The **App_Plugins** folder is the home for all third party packages installed on your site
+Depending on how you installed the plugin - Nuget vs The Backoffice, how you deploy your site to your production environment, and whether the plugin requires site specific configuration... will affect how you choose to version control a particular third party plugin. 
+
+** Nuget + Buildserver - if the plugin is installed via NuGet then as long as the packages.config file in the root of your project is source controlled, then the installed files for individual plugins shouldn't need to be source controlled (and your deployment process should pull the packages implementation files from Nuget during the deployment process).
+** Nuget + Manual deploy - here you don't necessarily need to source control the files, but you do need to remember to deploy them.
+** Backoffice - if your working with other developers, then it may be easier to add the plugin files to source control, or at least communicate to them that the particular package needs be installed via the backoffice
+** Backoffice + Buildserver - you'll need to include the plugin files in source control as the build server won't know to restore them - if the plugin/package doesn't come with a NuGet resource, consider setting up your own local NuGet repository for your build server to pull the files from.
+
+**Note**
+Each plugin could be different depending on what it's functionality is, and so may contain files it would be useful to track via Source control, and ones that should be ignored, check with the plugin's supporting website/developer for more information.
+
+### What folders and files should I **include** in my source control repository?
+
+#### Front-end build
+
+A lot depends on how you are running your front end build, eg you may be generating your css / js from SCSS or CoffeeScript, and it is the source of your frontend implementation that would need to be version controlled, allowing the actual css/js files to be generated during the build process, but generally speaking it's your Javascript, Css and Page Furniture eg /img and fonts - anything your site implementation uses statically that you would add to version control.
```

Comment:
> SCSS or CoffeeScript

What about less?  Is CoffeeScript still a thing?
https://trends.google.be/trends/explore?q=%2Fm%2F0hjc5m0,%2Fm%2F0gjd0jv,%2Fm%2F054k6n_,Scss